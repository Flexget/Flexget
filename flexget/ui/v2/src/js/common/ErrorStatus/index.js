import { connect } from 'react-redux';
import StatusBar from 'common/StatusBar';
import { clearStatus } from 'store/status/actions';

export function mapStateToProps({ status }) {
  return {
    open: !!status.error,
    message: status.error && status.error.message,
  };
}

export function mapDispatchToProps(dispatch) {
  return {
    clearStatus: () => dispatch(clearStatus()),
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(StatusBar);
