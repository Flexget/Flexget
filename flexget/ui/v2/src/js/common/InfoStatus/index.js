import { connect } from 'react-redux';
import StatusBar from 'common/StatusBar';
import { clearStatus } from 'store/status/actions';

export function mapStateToProps({ status }) {
  return {
    open: !!status.info,
    message: status.info,
  };
}

export function mapDispatchToProps(dispatch) {
  return {
    clearStatus: () => dispatch(clearStatus()),
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(StatusBar);
