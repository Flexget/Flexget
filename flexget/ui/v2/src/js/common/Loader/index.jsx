import React from 'react';
import PropTypes from 'prop-types';
import { CircularProgress } from 'material-ui/Progress';

const Loader = (props) => {
  if (props.isLoading) {
    // While our other component is loading...
    if (props.timedOut) {
      // In case we've timed out loading our other component.
      return <div>Loader timed out!</div>;
    } else if (props.pastDelay) {
      // Display a loading screen after a set delay.
      return <CircularProgress />;
    }
    // Don't flash "Loading..." when we don't need to.
    return null;
  } else if (props.error) {
    // If we aren't loading, maybe
    return <div>Error! Component failed to load</div>;
  }
  // This case shouldn't happen... but we'll return null anyways.
  return null;
};

Loader.propTypes = {
  isLoading: PropTypes.bool,
  timedOut: PropTypes.bool,
  pastDelay: PropTypes.bool,
  error: PropTypes.object, // eslint-disable-line react/require-default-props
};

Loader.defaultProps = {
  isLoading: false,
  timedOut: false,
  pastDelay: false,
};

export default Loader;
