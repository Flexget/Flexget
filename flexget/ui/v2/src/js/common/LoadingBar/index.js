import { connect } from 'react-redux';
import LoadingBar from './LoadingBar';

export function mapStateToProps({ status }, { prefix = '@flexget' } = {}) {
  return {
    loading: !!(Object.keys(status.loading).find(type => type.startsWith(prefix))),
  };
}

export default connect(mapStateToProps)(LoadingBar);
export { LoadingBar };
