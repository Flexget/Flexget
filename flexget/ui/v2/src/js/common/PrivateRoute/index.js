import { connect } from 'react-redux';
import PrivateRoute from './PrivateRoute';

export function mapStateToProps({ auth }) {
  return {
    loggedIn: !!auth.loggedIn,
  };
}

export default connect(mapStateToProps)(PrivateRoute);
export { PrivateRoute };
