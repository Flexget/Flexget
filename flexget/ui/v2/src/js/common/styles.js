import styled from 'react-emotion';
import theme from 'theme';
import Paper from 'material-ui/Paper';

export const PaperWrapper = styled(Paper)`
  padding: 2.4rem;
  display: flex;
  flex: 1;
  height: 100%;
  box-sizing: border-box;
  flex-direction: column;
  ${theme.breakpoints.up('sm')} {
    padding-top: 0;
  }
`;

export const Spacer = styled.div`
  flex: 1;
`;

export const NoPaddingWrapper = styled.div`
  height: calc(100% + 2rem);
  margin: -1rem;
  display: flex;
  flex-direction: column;
`;
