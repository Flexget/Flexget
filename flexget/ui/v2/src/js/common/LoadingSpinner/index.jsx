import React from 'react';
import PropTypes from 'prop-types';
import { Spinner, Wrapper } from './styles';

const LoadingBar = ({ loading }) => {
  if (loading) {
    return (
      <Wrapper>
        <Spinner />
      </Wrapper>
    );
  }
  return null;
};

LoadingBar.propTypes = {
  loading: PropTypes.bool,
};

LoadingBar.defaultProps = {
  loading: false,
};

export default LoadingBar;
