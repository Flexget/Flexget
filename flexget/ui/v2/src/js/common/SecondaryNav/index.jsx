import React from 'react';
import PropTypes from 'prop-types';
import { MuiThemeProvider } from 'material-ui/styles';
import { darkTheme } from 'theme';
import {
  SecondaryAppBar,
  SecondaryToolbar,
} from './styles';

const SecondaryNav = ({ children }) => (
  <MuiThemeProvider theme={darkTheme}>
    <SecondaryAppBar>
      <SecondaryToolbar>
        {children}
      </SecondaryToolbar>
    </SecondaryAppBar>
  </MuiThemeProvider>
);

SecondaryNav.propTypes = {
  children: PropTypes.node.isRequired,
};

export default SecondaryNav;
