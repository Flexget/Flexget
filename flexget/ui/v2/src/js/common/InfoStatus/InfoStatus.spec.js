import { mapStateToProps } from 'common/InfoStatus';

describe('plugins/common/InfoStatus', () => {
  it('should be correct if an info status should be displayed', () => {
    expect(mapStateToProps({ status: { info: 'Info Status' } })).toMatchSnapshot();
  });

  it('should be correct if an info status should not be displayed', () => {
    expect(mapStateToProps({ status: { } })).toMatchSnapshot();
  });
});
