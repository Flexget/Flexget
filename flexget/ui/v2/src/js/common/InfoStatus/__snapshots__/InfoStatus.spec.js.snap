// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`plugins/common/InfoStatus should be correct if an info status should be displayed 1`] = `
Object {
  "message": "Info Status",
  "open": true,
}
`;

exports[`plugins/common/InfoStatus should be correct if an info status should not be displayed 1`] = `
Object {
  "message": undefined,
  "open": false,
}
`;
